app/src/main/java/org/p2p/wallet/home/ui/main/HomePresenter.kt

```diff
@@ -463,27 +464,29 @@ class HomePresenter(
         userTokens: List<Token.Active>,
         ethTokens: List<Token.Eth>,
     ) {
-        Timber.d("local tokens change arrived")
-        state = state.copy(
-            tokens = userTokens,
-            ethTokens = ethTokens,
-            username = homeInteractor.getUsername(),
-        )
-        val isAccountEmpty = userTokens.all(Token.Active::isZero) && ethTokens.isEmpty()
-        when {
-            isAccountEmpty -> {
-                view?.showEmptyState(isEmpty = true)
-                handleEmptyAccount()
-            }
+        launch {
+            Timber.d("local tokens change arrived")
+            state = state.copy(
+                tokens = userTokens,
+                ethTokens = ethTokens,
+                username = homeInteractor.getUsername(),
+            )
+            val isAccountEmpty = userTokens.all(Token.Active::isZero) && ethTokens.isEmpty()
+            when {
+                isAccountEmpty -> {
+                    view?.showEmptyState(isEmpty = true)
+                    handleEmptyAccount()
```

Comment:
You can call `launch { handleEmptyAccount() }` instead of wrapping the entire function in `launch` call as there are no other suspend functions being called.
Alternatively, if the data is strongly related, you could add `suspend` keyword to the function, and let other functions decide in which coroutine it should run